pkg/middleware/middleware.go

```diff
@@ -125,6 +125,14 @@ func Process(quay *v1.QuayRegistry, obj client.Object, skipres bool) (client.Obj
 			return nil, err
 		}
 
+		if olabels := v1.GetLabelsOverrideForComponent(quay, kind); olabels != nil {
+			dep.Labels = olabels
```

Comment:
We should not fully override this property. I think we should do something like this:

```golang
for key, val := range olabels {
    dep.Labels[key] = val
}
```

Also, make sure `dep.Label` is not nil prior to this otherwise it will `panic`.